src/database/repositories/token_repository.py

```diff
@@ -119,3 +130,31 @@ def get_driver_profile(self,
                     raise InternalServer(str(e))
                 driver_profile = curs.fetchone()
         return DriverProfileTable.to_self(driver_profile)
+
+    def get_passenger_profile(self, token: bytes) -> DriverProfileTable:
+        query = f"""
+            SELECT passenger.*
+            FROM carmate.{self.POSTGRES_TABLE_NAME} tkn
+            INNER JOIN carmate.{UserRepository.POSTGRES_TABLE_NAME} usr 
+                ON tkn.user_id=usr.id
+            INNER JOIN carmate.{PassengerProfileRepository.POSTGRES_TABLE_NAME} passenger 
+                ON usr.id=passenger.user_id
+            WHERE tkn.token=%s
+        """
+
+        with establishing_connection() as conn:
+            with conn.cursor() as curs:
+                try:
+                    curs.execute(query, (token,))
+                except ProgrammingError:
+                    raise NotFound("passenger not found")
+                except IndexError:
+                    raise NotFound("passenger not found")
+                except Exception as e:
+                    raise InternalServer(str(e))
+                passenger_profile = curs.fetchone()
+
+            if passenger_profile is None:
+                raise NotFound("passenger not found")
```

Comment:
```suggestion
        if passenger_profile is None:
            raise NotFound("passenger not found")
```
Au même niveau que le return (je l'ai mal placé sur la dernière review)